packit_service/worker/handlers/testing_farm.py

```diff
@@ -171,7 +214,17 @@ def run_with_copr_builds(self, targets: List[str], failed: Dict):
             event_data["build_targets_override"] = list(targets_without_builds)
             self.run_copr_build_handler(event_data, len(targets_without_builds))
 
-        for target, copr_build in targets_with_builds.items():
+        if not targets_with_builds:
+            return
+
+        group = self._get_or_create_group(
+            list(targets_with_builds.keys()),
+            copr_build,
+            list(targets_with_builds.values()),
+        )
```

Comment:
it is not clear to me why we have to pass here the lastly assigned `copr_build`, which should be part of the `targets_with_builds` dictionary already; how does this work? :thinking:

Reply:
The second argument to `get_or_create_group` (in other words, the latest copr build) is used to get the `PipelineModel` to attach the tests to. I think that with copr groups, this `PipelineModel` is probably shared between all builds in `targets_with_builds` (they are probably part of the same group, i.e. attached to the same pipeline). This would mean that we can pick any build from `targets_with_builds`, I am just not completely sure if there can't be some edge-case where it would result in a different behavior. But it sounds like that the second argument is not necessary :thinking: